model/manifest/db.go

```diff
@@ -61,31 +63,36 @@ func FindFromVersion(versionID, project, revision, requester string) (*Manifest,
 	if err != nil {
 		return nil, errors.Wrap(err, "error finding manifest")
 	}
+	if manifest != nil {
+		return manifest, nil
+	}
 
 	// the version wasn't from the repotracker
+	// find the base commit's manifest
+	manifest, err = FindOne(ByBaseProjectAndRevision(project, revision))
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding manifest")
+	}
 	if manifest == nil {
-		manifest, err = FindOne(ByProjectAndRevision(project, revision))
+		return nil, nil
+	}
+
+	if evergreen.IsPatchRequester(requester) {
```

Comment:
This is now unreachable in most scenarios so I wanted to double check - this block is handling the scenario where your project has no modules, and you're patching adding the very first module to the config right?

Reply:
I think this block will run when a patch doesn't have its own manifest and we've retrieved the base commit's manifest from the db. This should be common: patches which don't change module definitions just use the base commit's manifest.

This block addresses a different issue: if a user specifies a SHA for a module when they add it in `set-module` we apply the changes to the module in memory in the manifest's `ModuleOverrides` without updating the module in the base commit's manifest in the db.

Reply:
Doh disregard, I misread the check above this